### PR TITLE
Fix notes correction diff truncation with GitHub-style hunk format

### PR DESCRIPTION
**Pull request by BASIC-BIT**
[AGENT]

Supersedes #133, this branch is rebased cleanly from `master`.

## Summary

- Replace the full unified diff in suggest-corrections with a compact GitHub-style hunk formatter
- Show changed lines with 3 lines of surrounding context by default
- Merge nearby changes into one hunk when context windows overlap or touch (including single-line gaps)

## Implementation

- Add shared formatter at `src/utils/diff.ts`
- Use it in Discord correction flow (`src/commands/notesCorrections.ts`)
- Use it in web correction flow (`src/trpc/routers/meetings.ts`)

## Testing

- Add robust tests in `test/utils/diff.test.ts` (27 total, including 7 hunk-merging edge cases)
- Run related suites:
  - `npx jest --findRelatedTests src/utils/diff.ts src/commands/notesCorrections.ts src/trpc/routers/meetings.ts --no-coverage`
- Full suite also ran green locally (`yarn test`)

Docs-exempt: internal behavior/formatting fix only.

### CODE REVIEW

File: src/utils/diff.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
+import { diffLines, type Change } from "diff";
+
+/** Lines of unchanged context shown above/below each changed region. */
+const DEFAULT_CONTEXT_LINES = 3;
+
+/** Separator between non-adjacent hunks. */
+const HUNK_SEPARATOR = "---";
```
[the remark this answers is not in the document]
Comment (BASIC-BIT) [reply src/utils/diff.ts]: [AGENT] Good catch. Updated the hunk separator to `@@ ... @@` so it is visually distinct from markdown horizontal rules inside note content, and added a regression test in `test/utils/diff.test.ts` (`keeps separator distinct from markdown horizontal rules in notes`).

File: src/utils/diff.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    if (lastVisibleIndex !== -1 && i - lastVisibleIndex > 1) {
+      output.push(HUNK_SEPARATOR);
+    }
+    lastVisibleIndex = i;
+
+    const { kind, text } = tagged[i];
+    const prefix = kind === "add" ? "+" : kind === "remove" ? "-" : " ";
+    output.push(`${prefix} ${text}`);
+
+    if (lineLimit && output.length >= lineLimit) break;
```
[the remark this answers is not in the document]
Comment (BASIC-BIT) [reply src/utils/diff.ts]: [AGENT] Agreed. Updated `formatHunkDiff` so `lineLimit` now applies only to diff content lines (`+`/`-`/context) and does not count separator lines. Also added a targeted test (`counts only content lines toward lineLimit, not separators`) to lock this behavior.

File: src/utils/diff.ts [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    output.push(`${prefix} ${text}`);
+    contentLines += 1;
+  }
+
+  let result = output.join("\n");
+
+  if (charLimit && result.length > charLimit) {
+    const suffix = "\n... (truncated)";
+    result = result.substring(0, charLimit - suffix.length) + suffix;
+  }
```
[the remark this answers is not in the document]
Comment (BASIC-BIT) [reply src/utils/diff.ts]: [AGENT] Great catch. I implemented this edge-case guard so `charLimit` is never exceeded when it is smaller than the truncation suffix length. The formatter now falls back to a plain prefix cut in that case, and I added `respects very small charLimit values` in `test/utils/diff.test.ts` to lock the behavior.
